components/802.15.4_RF/stm-s2lp-rf-driver/source/NanostackRfPhys2lp.cpp

```diff
@@ -572,6 +577,35 @@ static void rf_set_channel_configuration_registers(void)
     rf_write_register(PCKTCTRL5, preamble_len);
 }
 
+static void rf_configure_external_frontend()
+{
+    if (rf->ANT_SEL != NC) {
+        rf->ANT_SEL = MBED_CONF_S2LP_ANTENNA_SELECTION;
+
+        rf_write_register_field(S2LP_CSD_PIN, GPIO_FUNCTION_FIELD, TX_RX_INDICATION);
+        rf_write_register_field(S2LP_CSD_PIN, GPIO_MODE_FIELD, GPIO_DIGITAL_OUT_LOW);
+
+        rf_write_register_field(S2LP_CPS_PIN, GPIO_FUNCTION_FIELD, RX_INDICATION);
+        rf_write_register_field(S2LP_CPS_PIN, GPIO_MODE_FIELD, GPIO_DIGITAL_OUT_LOW);
+
+        rf_write_register_field(S2LP_CTX_PIN, GPIO_FUNCTION_FIELD, TX_STATE_INDICATION);
+        rf_write_register_field(S2LP_CTX_PIN, GPIO_MODE_FIELD, GPIO_DIGITAL_OUT_LOW);
+
+        tx_power_dbm = 10;
```

Comment:
Does this mean +10dBm when external PA is enabled. Would it be safer if the default is like 0 dBm and only increase it if user knows that the hardware can handle it.

Reply:
Yes, for the power amplifiers the maximum input power is 10dBm and s2lp max output power is 14dBm so I have to limit it so won't damage the device.